assembly/src/assembler/basic_block_builder.rs

```diff
@@ -1,22 +1,22 @@
 use alloc::{borrow::Borrow, string::ToString, vec::Vec};
 
-use vm_core::{mast::MastNodeId, AdviceInjector, AssemblyOp, Operation};
+use vm_core::{mast::MastNodeId, AdviceInjector, AssemblyOp, Decorator, Operation};
 
-use super::{
-    mast_forest_builder::MastForestBuilder, BodyWrapper, Decorator, DecoratorList, ProcedureContext,
-};
+use super::{mast_forest_builder::MastForestBuilder, BodyWrapper, DecoratorList, ProcedureContext};
 use crate::{ast::Instruction, AssemblyError, Span};
 
 // BASIC BLOCK BUILDER
 // ================================================================================================
 
-/// A helper struct for constructing SPAN blocks while compiling procedure bodies.
+/// A helper struct for constructing basic blocks while compiling procedure bodies.
 ///
-/// Operations and decorators can be added to a span builder via various `add_*()` and `push_*()`
-/// methods, and then SPAN blocks can be extracted from the builder via `extract_*()` methods.
+/// Operations and decorators can be added to a basic block builder via various `add_*()` and
+/// `push_*()` methods, and then basic blocks can be extracted from the builder via `extract_*()`
+/// methods.
 ///
-/// The same span builder can be used to construct many blocks. It is expected that when the last
-/// SPAN block in a procedure's body is constructed `extract_final_span_into()` will be used.
+/// The same basic block builder can be used to construct many blocks. It is expected that when the
+/// last basic block in a procedure's body is constructed [`Self::try_into_basic_block`] will be
+/// used.
 #[derive(Default)]
 pub struct BasicBlockBuilder {
```

Comment:
I suspect `BasicBlock` was chosen here to avoid conflict with `Block` which is defined in the same crate for the AST.

Reply:
IIRC it was a pretty arbitrary choice. If we do decide to switch to calling it `Block`, then we should change uses of "basic block" to "block" everywhere so that we're consistent (in another PR).